client/app/components/oui-message-alerter/oui-message-alerter.controller.js

```diff
@@ -0,0 +1,61 @@
+angular.module("App").controller("OuiMessageAlerterCtrl", class OuiMessageAlerterCtrl {
+
+    constructor ($state, ouiMessageAlerter) {
+        // Dependencies injection
+        this.$state = $state;
+        this.ouiMessageAlerter = ouiMessageAlerter;
+
+        this.messageTypes = ["error", "warning", "info", "success"]; // ordered by priority
+    }
+
+    /* =============================
+    =            EVENTS            =
+    ============================== */
+
+    onMessageDismissed (message) {
+        return this.ouiMessageAlerter.removeMessage(message);
+    }
+
+    /* -----  End of EVENTS  ------ */
+
+    /* =====================================
+    =            DISPLAY HELPER            =
+    ====================================== */
+
+    getMessagesByType (messageType) {
+        return this.ouiMessageAlerter.getMessagesFiltered({
+            type: messageType
+        }).reverse();
+    }
+
+    /* -----  End of DISPLAY HELPER  ------ */
+
+    /* =======================================
+    =            CONTROLLER HOOKS            =
+    ======================================== */
+
+    $onChanges (bindingsChanged) {
```

Comment:
Holy mackerel

Reply:
First time I use this hook :-)

Reply:
I feel like the code is really complicated but I don't know how you could improve it 😅